lib/reek/ast/node.rb

```diff
@@ -120,22 +105,22 @@ def source
       protected
 
       # See ".each_node" for documentation.
-      def look_for_type(target_type, ignoring = [], &blk)
+      def look_for(target_types, ignoring, &blk)
```

Comment:
I’m trying to understand why do we need both `look_for` and `look_for_recurse` – they only differ in that `look_for` returns only if the current node is _both_ of a target type _and_ ignored type, while `look_for_recurse` returns as soon as the current node’s type is one of the ignored ones. I tried to end up with just one of those two, but tests blew up. 😉 

Is there a way to conflate those two into one or is this an inescapable difference?